addons/osio-addon/src/main/java/io/fabric8/forge/generator/kubernetes/CreateBuildConfigStep.java

```diff
@@ -359,16 +359,18 @@ public Result execute(UIExecutionContext context) {
 
         if (addCI) {
             String discoveryNamespace = kubernetesClientHelper.getDiscoveryNamespace(jenkinsNamespace);
-            String jenkinsUrl = null;
-            try {
-                jenkinsUrl = KubernetesHelper.getServiceURL(kubernetesClientHelper.getKubernetesClient(), ServiceNames.JENKINS, jenkinsNamespace, "https", true);
-                discoveryNamespace = jenkinsNamespace;
-            } catch (Exception e) {
-                if (!discoveryNamespace.equals(jenkinsNamespace)) {
-                    try {
-                        jenkinsUrl = KubernetesHelper.getServiceURL(kubernetesClientHelper.getKubernetesClient(), ServiceNames.JENKINS, discoveryNamespace, "https", true);
-                    } catch (Exception e2) {
-                        throw new BadTenantException("Failed to find Jenkins URL in namespaces " + discoveryNamespace + " and " + jenkinsNamespace + ": " + e, e);
+            String jenkinsUrl = System.getenv("JENKINS_URL");
+            if (jenkinsUrl == null) {
+                try {
+                    jenkinsUrl = KubernetesHelper.getServiceURL(kubernetesClientHelper.getKubernetesClient(), ServiceNames.JENKINS, jenkinsNamespace, "https", true);
+                    discoveryNamespace = jenkinsNamespace;
+                } catch (Exception e) {
+                    if (!discoveryNamespace.equals(jenkinsNamespace)) {
+                        try {
+                            jenkinsUrl = KubernetesHelper.getServiceURL(kubernetesClientHelper.getKubernetesClient(), ServiceNames.JENKINS, discoveryNamespace, "https", true);
+                        } catch (Exception e2) {
+                            throw new BadTenantException("Failed to find Jenkins URL in namespaces " + discoveryNamespace + " and " + jenkinsNamespace + ": " + e, e);
```

Comment:
BadTenantException!? What's that for an exception Type?

Reply:
Should _jenkinsURL_ be validated?

Reply:
I know right? This code is a piece of :hankey: :)
I'll hopefully get it rewritten in a maintainable format soon. I'd like to avoid changing this part for now.

Reply:
> I know right?

Yeah, nothing to write home about ;-)